src/hooks/useKeyDown.tsx

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect, useCallback } from 'react';
+
+export default function useKeyDown(targetKey: string) {
+  // State for keeping track of whether key is pressed
+  const [keyPressed, setKeyPressed] = useState<boolean>(false);
+
+  // If pressed key is our target key then set to true
+  const downHandler = useCallback(
+    ({ key }: KeyboardEvent) => {
+      if (key === targetKey) {
+        setKeyPressed(true);
+      }
+    },
+    [targetKey]
+  );
+
+  // If released key is our target key then set to false
+  const upHandler = useCallback(
+    ({ key }: KeyboardEvent) => {
+      if (key === targetKey) {
+        setKeyPressed(false);
+      }
+    },
+    [targetKey]
+  );
+
+  // Add event listeners
+  useEffect(() => {
+    window.addEventListener('keydown', downHandler);
+    window.addEventListener('keyup', upHandler);
+    console.log('clicking ');
+    // Remove event listeners on cleanup
+    return () => {
+      window.removeEventListener('keydown', downHandler);
+      window.removeEventListener('keyup', upHandler);
+    };
+  }, [downHandler, upHandler]);
+
+  // If the user is not currently focused on the body, return
+  const activeEl = document.activeElement;
+  if (activeEl?.tagName !== 'BODY') return;
```

Comment:
let's return `false` here, since in the case afterwards `keyPressed` will be a boolean! will make for consistent return types